app/submitter/converter.py

```diff
@@ -82,17 +83,47 @@ def convert_answers_to_data(answer_store, questionnaire_schema):
     for answer in answer_store.answers:
         item = questionnaire_schema.get_item_by_id(answer['answer_id'])
         value = answer['value']
+
         if item is not None and value is not None:
-            if item.code not in data:
-                data[item.code] = _encode_value(value)
+            if not isinstance(item, CheckboxAnswer) or any('q_code' not in option for option in item.options):
+                data[item.code] = _get_answer_data(data, item.code, value)
             else:
-                if not isinstance(data[item.code], list):
-                    list_answers = [data[item.code]]
-                    data[item.code] = list_answers
-                data[item.code].append(_encode_value(value))
+                data.update(_convert_checkboxes_with_q_codes_to_data(item, value))
     return data
 
 
+def _get_answer_data(original_data, item_code, value):
+
+    if item_code in original_data:
+        data_to_return = original_data[item_code]
+        if not isinstance(data_to_return, list):
+            data_to_return = [data_to_return]
+        data_to_return.append(_encode_value(value))
+
+        return data_to_return
+    return _encode_value(value)
+
+
+def _convert_checkboxes_with_q_codes_to_data(item, value):
```

Comment:
It is a bit of a mouth full, but it is doing exactly what it says. I felt it would be slightly misleading if we used _get_checkbox_answer_data() as it would suggest all checkboxes use this, they don't its only the ones with q_codes in their options (UKIS)